src/Customer.php

```diff
@@ -338,6 +338,16 @@ public function addAddress(CustomerAddress $address)
         return $this;
     }
 
+    public function removeAddress($index)
+    {
+        if (array_key_exists($index, $this->addresses)) {
+            unset($this->addresses[$index]);
+            return true;
+        } else {
```

Comment:
No `else` is needed here, because of the return. 

Can you add type hints, e.g. int for the parameter index and bool for the method return value.